plugins/bundle/plugin.go

```diff
@@ -60,40 +70,105 @@ func Lookup(manager *plugins.Manager) *Plugin {
 // from the configured service. When a new bundle is downloaded, the data and
 // policies are extracted and inserted into storage.
 func (p *Plugin) Start(ctx context.Context) error {
-	p.logInfo("Starting bundle downloader.")
 	p.mtx.Lock()
 	defer p.mtx.Unlock()
-	p.downloader.Start(ctx)
+	for name, dl := range p.downloaders {
+		p.logInfo(name, "Starting bundle downloader.")
+		dl.Start(ctx)
+	}
 	return nil
 }
 
 // Stop stops the plugin.
 func (p *Plugin) Stop(ctx context.Context) {
-	p.logInfo("Stopping bundle downloader.")
 	p.mtx.Lock()
 	defer p.mtx.Unlock()
-	p.downloader.Stop(ctx)
+	for name, dl := range p.downloaders {
+		p.logInfo(name, "Stopping bundle downloader.")
+		dl.Stop(ctx)
+	}
 }
 
 // Reconfigure notifies the plugin that it's configuration has changed.
+// Any bundle configs that have changed or been added/removed will take
+// affect.
 func (p *Plugin) Reconfigure(ctx context.Context, config interface{}) {
-	p.mtx.Lock()
-	defer p.mtx.Unlock()
+	p.cfgMtx.Lock()
+	defer p.cfgMtx.Unlock()
 
+	// Look for any bundles that have had their config changed, are new, or have been removed
 	newConfig := config.(*Config)
-	if reflect.DeepEqual(p.config, *newConfig) {
-		p.logDebug("Bundle downloader configuration unchanged.")
+	deletedBundles := map[string]struct{}{}
```

Comment:
Nit: I think it would be good to factor the logic that diffs the config into a helper function (e.g., accept old and new config and return added, removed, updated configs.)

Reply:
Done.